src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -723,4 +803,360 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
+				enableAddRow: false, // TODO change when we support enableAddRow
+				enableAsyncPostRender: false,
+				editorFactory: {
+					getEditor: (column: ISlickColumn<any>) => this.getColumnEditor(column)
+				},
+				formatterFactory: {
+					getFormatter: this.getFormatter
+				}
+			};
+
+
+			if (dataSet.columnDefinitions) {
+
+				this.tables[0] = new Table(this.nativeElement.appendChild(newGridContainer), { dataProvider: this.gridDataProvider, columns: dataSet.columnDefinitions }, options);
+
+				for (let plugin of this.plugins[0]) {
+					this.tables[0].registerPlugin(plugin);
+				}
+
+				this._columnNameToIndex = {};
+				for (let i = 0; i < this.placeHolderDataSets[0].columnDefinitions.length; i++) {
+					this._columnNameToIndex[this.placeHolderDataSets[0].columnDefinitions[i].name] = i;
+				}
+			}
+		}
+		else {
+			this.tables[0] = new Table(this.nativeElement.appendChild(newGridContainer));
+		}
+	}
+
+
+
+	getOverridableTextEditorClass(): any {
+		let self = this;
+		class OverridableTextEditor {
+			private _textEditor: any;
+			public keyCaptureList: number[];
+
+			constructor(private _args: any) {
+				this._textEditor = new Slick.Editors.Text(_args);
+				const END = 35;
+				const HOME = 36;
+
+				// These are the special keys the text editor should capture instead of letting
+				// the grid handle them
+				this.keyCaptureList = [END, HOME];
+			}
+
+			destroy(): void {
+				this._textEditor.destroy();
+			}
+
+			focus(): void {
+				this._textEditor.focus();
+			}
+
+			getValue(): string {
+				return this._textEditor.getValue();
+			}
+
+			setValue(val): void {
+				this._textEditor.setValue(val);
+			}
+
+			loadValue(item, rowNumber): void {
+				if (self.overrideCellFn) {
+					let overrideValue = self.overrideCellFn(rowNumber, this._args.column.id, item[this._args.column.id]);
+					if (overrideValue !== undefined) {
+						item[this._args.column.id] = overrideValue;
+					}
+				}
+				this._textEditor.loadValue(item);
+			}
+
+			serializeValue(): string {
+				return this._textEditor.serializeValue();
+			}
+
+			applyValue(item, state): void {
+				let activeRow = self.currentCell.row;
+				let currentRow = self.dataSet.dataRows.at(activeRow);
+				let colIndex = self.getColumnIndex(this._args.column.name);
+				let dataLength: number = self.dataSet.dataRows.getLength();
+
+				// If this is not the "new row" at the very bottom
+				if (activeRow !== dataLength) {
+					currentRow[colIndex] = state;
+					this._textEditor.applyValue(item, state);
+				}
+			}
+
+			isValueChanged(): boolean {
+				return this._textEditor.isValueChanged();
+			}
+
+			validate(): any {
+				let activeRow = self.currentCell.row;
+				let result: any = { valid: true, msg: undefined };
+				let colIndex: number = self.getColumnIndex(this._args.column.name);
+				let newValue: any = this._textEditor.getValue();
+
+				// TODO: It would be nice if we could support the isCellEditValid as a promise
```

Comment:
what's the plan with the TODO?  is this something we'll be doing soon-ish?  otherwise, I don't see a lot of value in this type of comment (i.e. if it's not important enough to do now who will ever have time to do this in the future?)

Reply:
Removed